packages/dotnet/analyzer/Utilities/TargetBuilder.cs

```diff
@@ -0,0 +1,560 @@
+                $"{outputPrefix}/{intermediatePath}"
+            ],
+            Metadata = new TargetMetadata
+            {
+                Description = "Build the .NET project in Release configuration",
+                Technologies = ProjectUtilities.GetTechnologies(fileName, isTest)
+            }
+        };
+    }
+
+    private static void AddTestTarget(
+        Dictionary<string, Target> targets,
+        string projectName,
+        string fileName,
+        List<PackageReference> packageRefs,
+        Dictionary<string, string> properties,
+        string workspaceRoot)
+    {
+        // TODO(@AgentEnder): We should add this back in after external deps
+        // support is fleshed out.
+        //
+        // var externalDeps = packageRefs.Select(p => p.Include).ToArray();
+        var testResultsDir = GetTestResultsDirectory(properties, projectName, workspaceRoot);
+
+        var useWorkspaceRoot = UsesArtifactsOutput(properties);
+        var outputPrefix = useWorkspaceRoot ? "{workspaceRoot}" : "{projectRoot}";
+
+        string[] defaultFlags = ["--no-build", "--no-restore"];
+
+        targets["test"] = new Target
+        {
+            Command = "dotnet test",
+            Options = new TargetOptions
+            {
+                Cwd = "{projectRoot}",
+                Args = [.. defaultFlags]
+            },
+            DependsOn = ["build", "restore"],
+            Cache = true,
+            Inputs =
+            [
+                "default",
+                "^production",
+                // new { externalDependencies = externalDeps }
+            ],
+            Outputs = [$"{outputPrefix}/{testResultsDir}"],
+            Metadata = new TargetMetadata
+            {
+                Description = "Run .NET tests",
+                Technologies = ProjectUtilities.GetTechnologies(fileName, true)
+            }
+        };
+    }
+
+    private static void AddRestoreTarget(
+        Dictionary<string, Target> targets,
+        string projectName,
+        string fileName,
+        bool isTest,
+        Dictionary<string, string> properties,
+        string workspaceRoot)
+    {
+        var intermediatePath = GetIntermediateOutputPath(properties, projectName, workspaceRoot);
+
+        var useWorkspaceRoot = UsesArtifactsOutput(properties);
+        var outputPrefix = useWorkspaceRoot ? "{workspaceRoot}" : "{projectRoot}";
+
+        targets["restore"] = new Target
+        {
+            Command = "dotnet restore",
+            Options = new TargetOptions
+            {
+                Cwd = "{projectRoot}",
+                Args = ["--no-dependencies"]
+            },
+            DependsOn = ["^restore"],
+            Cache = true,
+            Inputs =
+            [
+                "{projectRoot}/*.csproj",
+                "{projectRoot}/*.fsproj",
+                "{projectRoot}/*.vbproj"
+            ],
+            Outputs = [$"{outputPrefix}/{intermediatePath}"],
```

Comment:
double check this?

Reply:
Yeah, I was afraid of it. This isn't accurate... restore's outputs to a global location a'la npm's cache or the .cargo directory.... so I don't think we can actually cache this.

We should be fine, but going to remove the dependency on restore and people will just have to be responsible for making sure packages are restored.